tutorials/epochs/plot_40_autogenerate_metadata.py

```diff
@@ -0,0 +1,449 @@
+             'stimulus_side'] = 'left'
+
+# right-side stimulation
+metadata.loc[metadata['stimulus'].isin(['compatible/target_right',
+                                        'incompatible/target_right']),
+             'stimulus_side'] = 'right'
+
+# first assume all responses were incorrect, then mark those as correct where
+# the stimulation side matches the response side
+metadata['response_correct'] = False
+metadata.loc[metadata['stimulus_side'] == metadata['response'],
+             'response_correct'] = True
+
+metadata
+
+###############################################################################
+# Now it's already time to epoch the data! When deciding upon the epochs
+# duration for this specific analysis, we need to ensure we see quite a bit of
+# signal from before and after the motor response. We also must be aware of
+# the fact that motor-/muscle-related signals will most likely be present
+# **before** the response button trigger pulse appears in our data, so the time
+# period close to the response event should not be used for baseline
+# correction. But at the same time, we don't want to use a baseline
+# period that extends too far away from the button event. The following values
+# seem to work quite well.
+
+epochs_tmin, epochs_tmax = -0.6, 0.4
+baseline = (-0.4, -0.2)
+reject = {'eeg': 250e-6}
+epochs = mne.Epochs(raw=raw, tmin=epochs_tmin, tmax=epochs_tmax,
+                    baseline=baseline, reject=reject,
+                    events=events, event_id=event_id, metadata=metadata,
+                    preload=True)
+
+###############################################################################
+# Let's do a final sanity check: we want to make sure that in every row, we
+# actually have a stimulus. We use ``epochs.metadata`` (and not ``metadata``)
+# because when creating the epochs, we passed the ``reject`` parameter, and
+# MNE-Python always ensures that ``epochs.metadata`` stays in sync with the
+# available epochs.
+
+epochs.metadata.loc[epochs.metadata['stimulus'].isnull(), :]
+
+###############################################################################
+# Bummer! ☹️ It seems the very first two responses were recorded before the
+# first stimulus appeared: the values in the ``stimulus`` column are ``None``.
+# There is a very simple way to select only those epochs that **do** have a
+# stimulus (i.e., are not ``None``):
+
+epochs = epochs['stimulus']
+
+###############################################################################
+# Time to calculate the ERPs for correct  and incorrect responses.
+# For visualization, we'll only look at sensor ``FCz``, which is known to show
+# the ERN nicely in the given paradigm. We'll also create a topoplot to get an
+# impression of the average scalp potentials measured in the first 100 ms after
+# an incorrect response.
+
+resp_erp_correct = epochs['response_correct == True'].average()
+resp_erp_incorrect = epochs['response_correct == False'].average()
```

Comment:
comparing to `bool` with `==`?  Why not `epochs['response_correct']` and `epochs['~response_correct']`

Reply:
I know this is bad practice, but I felt this was "more readable" for the `False` part. But since we shouldn't teach bad practices, I fully embrace your proposal :)

Reply:
you can use `not response_correct` instead of `~response_correct` if readability is a concern.